endpoint.cpp

```diff
@@ -346,35 +349,50 @@ void Endpoint::log_aggregate(unsigned int interval_sec)
     }
 }
 
-int UartEndpoint::open(const char *path, speed_t baudrate)
+UartEndpoint::~UartEndpoint()
+{
+    if (_change_baud_timeout) {
+        Mainloop::get_instance().del_timeout(_change_baud_timeout);
+        _change_baud_timeout = nullptr;
+    }
+}
+
+int UartEndpoint::open(const char *path, unsigned long baudrates[], size_t baud_size)
```

Comment:
No.  set_speed() being public doesn't mean it needs to be called from outside.  You keep the timer here but but the timer callback just calls set_speed(), dealing with updating the `next_speed`

I'm waiting for that PR to be reviewed.... but yes, would be better to rebase.